org.testeditor.web.backend.xtext/src/test/java/org/testeditor/web/backend/xtext/index/IndexServiceClientTest.xtend

```diff
@@ -321,36 +285,33 @@ class IndexServiceClientTest {
 		]
 	}
 
-	private def mockedIndexService(String payload, List<IEObjectDescription> resultingEObjectDescriptions) {
-		val request = mock(HttpServletRequest)
-		when(request.getHeader(eq(AUTHORIZATION))).thenReturn(AUTH_HEADER)
+	private def void setupMocksWithAuthHeader(String payload, List<IEObjectDescription> resultingEObjectDescriptions) {
+		setupMocks(payload, resultingEObjectDescriptions)
+		when(contextRequest.getHeader(eq(AUTHORIZATION))).thenReturn(AUTH_HEADER)
+	}
 
-		return this.mockedIndexService(payload, resultingEObjectDescriptions, request)
+	private def void setupMocks(String payload, List<IEObjectDescription> resultingEObjectDescriptions) {
+		when(requestProvider.get).thenReturn(contextRequest)
+		setupMocksWithoutRequest(payload, resultingEObjectDescriptions)
 	}
 
-	private def mockedIndexService(String payload, List<IEObjectDescription> resultingEObjectDescriptions,
-		HttpServletRequest contextRequest) {
-		val client = mock(Client)
+	private def void setupMocksWithoutRequest(String payload, List<IEObjectDescription> resultingEObjectDescriptions) {
 		val target = mock(WebTarget)
 		val invocationBuilder = mock(Builder)
-		when(client.target(any(URI))).thenReturn(target)
+		when(client.target(eq(uri))).thenReturn(target)
 		when(target.queryParam(any, any)).thenReturn(target)
 		when(target.request(anyString)).thenReturn(invocationBuilder)
 		when(invocationBuilder.header(eq(AUTHORIZATION), eq(AUTH_HEADER))).thenReturn(invocationBuilder)
 
 		val payloadMatcher = if(payload === null) any else eq(Entity.text(payload))
 		when(invocationBuilder.post(payloadMatcher, any(GenericType))).thenReturn(resultingEObjectDescriptions)
-
-		val uri = URI.create("http://example.org")
-
-		return new IndexServiceClient(client, uri, [contextRequest])
 	}
 
 	private def mockedResource(String payload, EList<EObject> resourceContents) {
 		val resource = mock(XtextResource)
 		val serializer = mock(ISerializer)
 		when(resource.resourceSet).thenReturn(mock(XtextResourceSet))
-		when(resource.URI).thenReturn(mock(org.eclipse.emf.common.util.URI))
+		when(resource.URI).thenReturn(mock(URI))
```

Comment:
please add a hint about mocking final classes with mockito-2
https://github.com/mockito/mockito/wiki/What%27s-new-in-Mockito-2#mock-the-unmockable-opt-in-mocking-of-final-classesmethods

Reply:
Put a comment block at the top of the file, above the declaration of the field `uri`.